osquery/sql/sqlite_network.cpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <string>
+
+#include <boost/asio/ip/address.hpp>
+#include <boost/asio/ip/network_v4.hpp>
+#include <boost/asio/ip/network_v6.hpp>
+#include <sqlite3.h>
+
+namespace errc = boost::system::errc;
+namespace ip = boost::asio::ip;
+
+namespace osquery {
+
+static void sqliteCidrRangeFunc(sqlite3_context* context,
+                                int argc,
+                                sqlite3_value** argv) {
+  const size_t cidr_idx = 0, ipaddr_idx = 1;
+  if (sqlite3_value_type(argv[cidr_idx]) != SQLITE_TEXT) {
+    sqlite3_result_error(context, "CIDR must be a string", -1);
+    return;
+  }
+
+  if (sqlite3_value_type(argv[ipaddr_idx]) != SQLITE_TEXT) {
+    sqlite3_result_error(context, "IP addresss must be a string", -1);
+    return;
+  }
+
+  const char* cidr_str =
+      reinterpret_cast<const char*>(sqlite3_value_text(argv[cidr_idx]));
+  const char* ipaddr_str =
+      reinterpret_cast<const char*>(sqlite3_value_text(argv[ipaddr_idx]));
+
+  boost::system::error_code ec;
+  ip::address ipaddr = ip::make_address(ipaddr_str, ec);
+  if (ec.value() != errc::success) {
+    sqlite3_result_error(context, "IP address cannot be parsed", -1);
+    return;
+  }
+
+  if (ipaddr.is_v4()) {
+    ip::network_v4 network = ip::make_network_v4(cidr_str, ec);
+    if (ec.value() != errc::success) {
+      sqlite3_result_error(
+          context, "CIDR for IP address v4 cannot be parsed", -1);
+      return;
+    }
+
+    ip::address_v4_range all_network_hosts = network.hosts();
+    bool is_in_range =
+        all_network_hosts.find(ipaddr.to_v4()) != all_network_hosts.end();
```

Comment:
i'm having trouble understanding what boost is doing here. Is this efficient for large network blocks?

I see internet discussion about a couple of ways, but nothing in the boost docs

Reply:
Testing this PR, this appear efficient, so I assume boost is doing something reasonable under the hood.